client/my-sites/people/edit-team-member-form/edit-user-form.jsx

```diff
@@ -77,24 +77,31 @@ class EditUserForm extends React.Component {
 	}
 
 	getAllowedSettingsToChange() {
-		const { currentUser, user, isJetpack } = this.props;
+		const { currentUser, user, isJetpack, hasWPCOMAccountLinked } = this.props;
 		const allowedSettings = [];
 
 		if ( ! user.ID ) {
 			return allowedSettings;
 		}
 
-		// On WP.com sites, a user should only be able to update role.
-		// A user should not be able to update own role.
+		// On any site, admins should be able to change only other
+		// user's role or isExternalContributor.
 		if ( isJetpack ) {
+			// Jetpack self hosted or Atomic.
 			if ( ! user.linked_user_ID || user.linked_user_ID !== currentUser.ID ) {
 				allowedSettings.push( 'roles', 'isExternalContributor' );
 			}
-			allowedSettings.push( 'first_name', 'last_name', 'name' );
 		} else if ( user.ID !== currentUser.ID ) {
+			// WP.com Simple sites.
 			allowedSettings.push( 'roles', 'isExternalContributor' );
 		}
 
+		// On any site, allow editing 'first_name', 'last_name', 'name'
+		// only for users without WP.com account.
+		if ( ! hasWPCOMAccountLinked ) {
+			allowedSettings.push( 'first_name', 'last_name', 'name' );
```

Comment:
Just an observation. Seeing as though the individual setting name, the field name and the field id, and the `recordFieldFocus` are all the same it would make sense to centralise these values in an Object or similar rather than relying on string values outside of TypeScript where we'd have that type safety.

Reply:
Yep I think creating a maintenance issue for this might be a good idea so we don't forget it and can follow up.

Reply:
Added it here https://github.com/Automattic/wp-calypso/issues/53252. I will pick it up as part of my 20%, TS rulez!